### ISSUE TITLE
Raise snackbar notifications on Favorite actions

### ISSUE THREAD

**Issue by DenysVuika** [Alfresco/alfresco-content-app#953]
**Is your feature request related to a problem? Please describe.**

The only way to see if a library (or any other node) has been successfully favorited/unfavorited is to invoke the menu (toolbar, context) once again to see that the "star" icon changed.
This behavior is not very user-friendly.

**Describe the solution you'd like**
Every time a favorite/unfavorite action is performed, the user should see a corresponding Snackbar (info or error).

**Describe alternatives you've considered**

No alternatives

**Additional context**

No extra context

**Comment by binduwavell** [comment]
What about showing favorite flag in the doc list row? A customizable flagging feature for other things would be pretty killer too. Let’s say we build something to OCR docs would be nice to have a flag showing OCR status. Or if digital signature is requested, would be nice to show state of request and ultimately that the doc is signed.

**Comment by DenysVuika** [comment]
That's why I raised the #935 ticket. To change the icons we need some mechanism to dynamically change/evaluate icon values.

**Comment by binduwavell** [comment]
I’m a bit confused. In Share there is the concept of flags (small icons) next to the larger thumbnail. These allow you to visually see in a list view which items are locked, favorited, have EXIF metadata, etc. It is possible to create a customization to add a new type of flag. 

If we added a similar idea to ADF, your other proposal could be used to have a flag show as an empty star for things that are not favorited and a full star for things that are favorited. 

Although, I think not showing the flag for things that are not favorited would be cleaner. 

If this is created as a general extensible capability, it would be cool to be able to configure an action to fire when a flag is clicked. 

This would allow the favorite flag to be clicked to unfavorite. 

Or if we have a flag to show docs with tags, clicking could launch a tag editing dialog...

I think there are tons of good use cases for this UI capability...

**Comment by binduwavell** [comment]
Grr s/ADF/ACA/g

**Comment by DenysVuika** [comment]
@binduwavell I agree regarding overlays/flags on images. Am thinking over a separate and more sophisticated thumbnail component to use as a column template instead of the default one.

**Comment by binduwavell** [comment]
Could be overlay on thumbnail or flags could be a reactive component with small icons that stack 4x4 next to a thumbnail, but could have less or more and would show as a grid of small icons that grows shrinks based on the number of flags that need to be displayed.
